tensorboard/plugins/profile/memory_viewer/memory_usage/memory-usage.ts

```diff
@@ -94,7 +94,31 @@ namespace memory_viewer_usage {
       }
       this.initBuffers_(bufferAssignment);
       this.initAllocations_(bufferAssignment);
-      this.findPeakMemoryUsage_(bufferAssignment);
+      const trace = this.getHbmHeapTrace_(bufferAssignment);
+      if (!trace) {
+        console.error('Missing hbm heap simulator trace.');
+        return;
+      }
+      this.findPeakMemoryUsage_(trace);
+    }
+
+    /**
+     * From a list of heap simulator traces, identify the one uses the HBM
+     * memory space.
+     */
+    private getHbmHeapTrace_(bufferAssignment) {
+      const kHbmColor = 0;
+      for (const trace of bufferAssignment.heapSimulatorTraces) {
+        for (const event of trace.events) {
+          if (!event.bufferId) continue;
+          const buffer = this.idToBuffer_[event.bufferId];
+          if (!buffer) continue;
+          if (buffer.color != kHbmColor) break;
```

Comment:
Not that I understand fully what is going on but finding a trace based on color seems very odd. Can you help me understand? Thanks!

Reply:
The color here does not mean the visual color, it is a compiler term used in graph coloring for memory spaces.
0 means the memory region is allocated in device memory. 1 means vmem and 2 means cmem.  (I think that's confusing here, added some comments.)